packages/compartment-mapper/src/import-hook.js

```diff
@@ -156,6 +158,21 @@ export const makeImportHookMaker = (
           // Archived compartments are not executed.
           return freeze({ imports: [], exports: [], execute() {} });
         }
+        if (exitModuleImportHook) {
+          console.error('#################i');
+          const record = await exitModuleImportHook(moduleSpecifier);
+          if (record) {
+            // It'd be nice to check the policy before importing it, but we can only throw a policy error if the
+            // hook returns something. Otherwise, we need to fall back to the 'cannot find' error below.
+            enforceModulePolicy(moduleSpecifier, compartmentDescriptor, {
+              exit: true,
+            });
+            // note it's not being marked as exit in sources
+            // it could get marked and the second pass, when the archive is being executed, would have the data
+            // to enforce which exits can be dynamically imported
+            return record;
```

Comment:
Introducing attenuation by policy in here is going to be difficult, because each time a builtin is imported it'd have to get a new instance for different attenuations and here it'll get memoized.